src/TimeWarpEventDispatcher.cpp

```diff
@@ -88,28 +91,42 @@ void TimeWarpEventDispatcher::startSimulation(const std::vector<std::vector<Logi
         threads.push_back(std::move(thread));
     }
 
-    unsigned int gvt = 0;
-    auto sim_start = std::chrono::steady_clock::now();
 
-    // Master thread main loop
-    while (!termination_manager_->terminationStatus()) {
-
-        comm_manager_->handleMessages();
+    pthread_barrier_init(&termination_barrier_sync_1, NULL, num_worker_threads_+2);
```

Comment:
please explain the logic for num_worker_threads_ + 2

Reply:
We now have three manager threads, one for gvt calculations, one for the communication manager, and one for termination. So we need two extra threads for those manager functions.